### PR TITLE
[ONNX] Deprecate setter functions for global variables

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85488
* #85487
* #85486
* #85350
* #85349
* #85312
* __->__ #85165
* #85491
* #84776
* #84728

`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

### CODE REVIEW

File: torch/_C/__init__.pyi.in [hunk 1]
```diff
@@ -404,7 +404,7 @@ def _jit_pass_onnx_block(
     is_sub_block: _bool
 ) -> Dict[Value, Value]: ...
 def _jit_pass_onnx_assign_scoped_names_for_node_and_value(graph: Graph) -> None: ...
-def _jit_pass_fixup_onnx_controlflow_node(n: Node, opset_version: _int) -> Node: ...
+def _jit_pass_fixup_onnx_controlflow_node(n: Node, opset_version: _int) -> List[Value]: ...
```
Comment (BowenBao) [inline torch/_C/__init__.pyi.in:407]: Please add this fix to description too.

File: torch/onnx/symbolic_helper.py [hunk 2]
```diff
@@ -1645,16 +1651,31 @@ def args_have_same_dtype(args):
 
 
 # TODO(justinchuby): Delete these setters, users should set the vars directly.
+@_deprecation.deprecated(
+    "1.13.0",
+    "TBD",
```
Comment (BowenBao) [inline torch/onnx/symbolic_helper.py:1656]: Why not set it to "1.14.0" as well?

**Review by BowenBao — APPROVED** [review]
LG, minor comments.

File: torch/_C/__init__.pyi.in (hunk 1 above)
Comment (justinchuby) [reply torch/_C/__init__.pyi.in]: Done

File: torch/onnx/symbolic_helper.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/symbolic_helper.py:1656]: Sure thing!

File: torch/onnx/_globals.py [hunk 3]
```diff
@@ -28,7 +27,9 @@ def __init__(self):
         self._in_onnx_export: bool = False
         # Whether the user's model is training during export
         self.export_training: bool = False
-        self.operator_export_type: Optional[_C_onnx.OperatorExportTypes] = None
+        self.operator_export_type: _C_onnx.OperatorExportTypes = (
```
Comment (AllenTiTaiWang) [inline torch/onnx/_globals.py:30]: I am planning on adding a new type (https://github.com/pytorch/pytorch/issues/84884). Any suggestion for that?

**Review by titaiwangms — APPROVED** [review]
LGTM!

File: torch/onnx/_globals.py (hunk 3 above)
Comment (justinchuby) [reply torch/onnx/_globals.py:30]: Addressed offline

**Pull request closed without merging**